### DISCUSSION TITLE
[lint] unsupported TypeScript types

### DISCUSSION THREAD

**Discussion by Conaclos** [rome/tools#1640] [Suggestions]
TypeScript `lib.d.ts` exports several types. Some types are not currently supported by `rome check`.

for instance, `rome check` complains about any (typed) array constructor (Uint8ArrayConstructor, ...).

```ts
  ✖ The Uint8ArrayConstructor variable is undeclared

    4 │ export const b = a + 1
    5 │ 
  > 6 │ export type X = Uint8ArrayConstructor
      │                 ^^^^^^^^^^^^^^^^^^^^^
```

**Comment by ematipico** [comment]
Are there any other types that we don't support? At the moment [we use non-optimal solution](https://github.com/rome/tools/blob/main/internal/compiler/lint/rules/js/noUndeclaredVariables.ts#L37-L72) to track them.

**Reply by Conaclos** [reply]
@ematipico
Yeah a lot are missing. Basically every Constructor type, ...
I can do a PR if you are ok.

**Reply by Conaclos** [reply]
I submitted a PR: #1646
